File: S5/HLIN505 - Modélisation et Programmation par Objets 2/TP/2016-1017/TP3/Pokemon/src/Pokemon.java
public abstract class Pokemon {
	protected String name;
	protected double poids;
	public Pokemon(String name , double poids) {
		this.name = name;
		this.poids = poids;
	}
	public abstract double vitesse();
	public String toString() {
		return "Je suis le Pokémon " + name + ", mon poids est de " + poids + "kg, ma vitesse est de " + vitesse() + "m/s ";
	}
}

File: S6/HLIN603 - Objets avances/TPs/2016-2017/TP1/Exercice 2/source/Pokemon.cc
#include "../header/Pokemon.h"
#include <sstream>
#include <iostream>

Pokemon::Pokemon(std::string s, int n) : nom(s) , poids(n) {}

std::string Pokemon::toString() {

  std::stringstream ss;
  std::string poids;
  ss << this->poids;
  ss >> poids;

  return "Je suis le Pokemon " + this->nom + ", mon poids est de " + poids + "kg, ";
}
